core/src/main/java/org/commonjava/maven/ext/core/impl/RangeResolver.java

```diff
@@ -79,181 +83,234 @@ public void init( final ManipulationSession session )
             }
-            catch ( RuntimeException e )
+            catch ( final RuntimeException e )
             {
                 if ( e.getCause() instanceof ManipulationException )
                 {
-                    throw (ManipulationException)e.getCause();
+                    throw ( ManipulationException ) e.getCause();
                 }
                 throw e;
             }
         }
         return changed;
     }
 
-    private void handleVersionWithRange( Plugin p )
+    private void handleVersionWithRange( final Plugin plugin, final PropertyInterpolator pi )
+            throws ManipulationException
     {
+        final String version = pi.interp ( plugin.getVersion() );
+
+        if ( StringUtils.isEmpty( version ) )
+        {
+            return;
+        }
+
         try
         {
-            VersionRange versionRange = VersionRange.createFromVersionSpec( p.getVersion() );
+            final VersionRange versionRange = VersionRange.createFromVersionSpec( version );
 
-            // If its a range then try to use a matching version...
+            // If it's a range then try to use a matching version...
             if ( versionRange.hasRestrictions() )
             {
-                final List<ArtifactVersion> versions = getVersions( new SimpleProjectRef( p.getGroupId(), p.getArtifactId() ) );
+                final String groupId = pi.interp( plugin.getGroupId() );
+                final String artifactId = pi.interp( plugin.getArtifactId() );
+                final ProjectRef ref = new SimpleProjectRef( groupId, artifactId );
+                final List<ArtifactVersion> versions = getVersions( ref );
                 final ArtifactVersion result = versionRange.matchVersion( versions );
 
-                logger.debug( "Resolved range for plugin {} got versionRange {} and potential replacement of {}", p, versionRange, result );
+                logger.debug( "Resolved range for plugin {} got versionRange {} and potential replacement of {}",
+                        plugin, versionRange, result );
 
                 if ( result != null )
                 {
-                    p.setVersion( result.toString() );
+                    plugin.setVersion( result.toString() );
                 }
                 else
                 {
                     logger.warn( "Unable to find replacement for range." );
                 }
             }
         }
-        catch ( InvalidVersionSpecificationException e )
+        catch ( final InvalidVersionSpecificationException | ManipulationException e )
         {
             throw new ManipulationUncheckedException( new ManipulationException( "Invalid range", e ) );
         }
     }
 
-    private void handleVersionWithRange( Dependency d )
+    private void handleVersionWithRange( Dependency dependency, PropertyInterpolator pi ) throws ManipulationException
     {
+        final String version = pi.interp( dependency.getVersion() );
```

Comment:
`PropertyResolver` already does this and handles properties defined within a parent (but still within the project)

There is also the functions on Project that can return a fully resolved list of Deps/Plugins/etc handling resolving as well.

I think we should reuse some of that instead which would also mean you wouldn't need to pass a `PropertyInterpolator` around and simplify some of these changes.